### PR TITLE
Cookie expiration config argument

### PR DESCRIPTION
**Pull request by phoward38**
- Add cookie_expiration argument to global_sessions_config w/ same default & 
  - Option for NULL (session cookie)

@merlinoa I updated the same in `set_password_module.js` but didn't push as I wasn't sure where it's being called

### CODE REVIEW

File: R/global_sessions_config.R [hunk 1]
```diff
@@ -29,6 +29,8 @@
 #' allow user to do certain actions (such as viewing charts and tables) without signing in,
 #' and only require users to sign in if they want to save data to your database.
 #' @param sentry_dsn either \code{NULL}, the default, or your Sentry project DSN.
+#' @param cookie_expiration the number of days before a User's cookie expires.
+#' Set to `NULL` to force Sign Out at session end. Default is `365` (i.e. 1 year)
```
Comment (merlinoa) [inline R/global_sessions_config.R:33]: use \code{NULL} not `NULL`

File: R/global_sessions_config.R [hunk 2]
```diff
@@ -29,6 +29,8 @@
 #' allow user to do certain actions (such as viewing charts and tables) without signing in,
 #' and only require users to sign in if they want to save data to your database.
 #' @param sentry_dsn either \code{NULL}, the default, or your Sentry project DSN.
+#' @param cookie_expiration the number of days before a User's cookie expires.
```
Comment (merlinoa) [inline R/global_sessions_config.R:32]: don't capitalize User's

File: R/global_sessions_config.R [hunk 3]
```diff
@@ -56,13 +58,18 @@ global_sessions_config <- function(
   sign_in_providers = "email",
   is_email_verification_required = TRUE,
   is_auth_required = TRUE,
-  sentry_dsn = NULL
+  sentry_dsn = NULL,
+  cookie_expiration = 365L
```
Comment (merlinoa) [inline R/global_sessions_config.R:62]: can we call it "cookie_expires" to match up better with the option in js-cookie

File: R/global_sessions_config.R (hunk 2 above)
Comment (merlinoa) [reply R/global_sessions_config.R:32]: Also in the docs.  mention "This argument is passed to the "expires" option in js-cookie: \url{https://github.com/js-cookie/js-cookie#expires}"

File: R/sign_in_components.R [hunk 4]
```diff
@@ -26,7 +26,7 @@ sign_in_js <- function(ns) {
       firebase_dependencies(),
       firebase_init(firebase_config),
       tags$script(src = "polish/js/auth_firebase.js?version=4"),
```
Comment (merlinoa) [inline R/sign_in_components.R:28]: bump the version to 5

**Review by merlinoa — CHANGES_REQUESTED** [review]
Looks good.  Please update per my comments.  Also can you please test with a negative number and a number with a decimal (e.g. 0.5).  We probably should not allow negative numbers.  If non integer numbers work, then they should be allowed.  Submit a another PR with updates.

File: R/sign_in_components.R (hunk 4 above)
Comment (phoward38) [reply R/sign_in_components.R]: 6e55a79f72c8e8e49ab74c2a3da1297c7f450763

File: R/global_sessions_config.R (hunk 1 above)
Comment (phoward38) [reply R/global_sessions_config.R]: c0629b28e77a52a0cb52ba35abf2f0ea50cfef95

File: R/global_sessions_config.R (hunk 3 above)
Comment (phoward38) [reply R/global_sessions_config.R]: e331a9d73c1be00ac5890acf54976c81ced54bd5

**Comment by phoward38** [comment]
@merlinoa The `cookie_expires` value cannot be less than or equal to 0, but it can be a real number or integer. I updated the argument validation: 0c21c41168721f44acf1ead8e009f01439a092f0

**Review by merlinoa — APPROVED** [review]
